### PR TITLE
fix(Types): Fix typings

### PR DESCRIPTION
**Pull request by PalmDevs**
This PR fixes typings for what was initially done.

### CODE REVIEW

**Review by axisiscool — APPROVED** [review]

**Comment by axisiscool** [comment]
What was done initially wasn't correct at all. I can do typing improvements soon enough.

File: src/util/Types.ts [hunk 1]
```diff
@@ -23,17 +23,16 @@ export type ChannelResolvable = Channel | string;
 export type UserResolvable = User | ClientUser | BaseUser | string;
 export type EventResolvable = 'ready' | 'waitingForGuilds' | 'guildAvailable' | 'guildCreate' | 'guildUpdate' | 'message';
 export type ActivityStatus = 'online' | 'idle' | 'dnd' | 'invisible';
-
 export type MessageContent = string | Embed | {
     content?: string;
-    embeds?: [ Embed ],
+    embeds?: Partial<Embed>[],
```
Comment (axisiscool) [inline src/util/Types.ts]:
```suggestion
    embeds?: Embed[],
```
The partial type makes no sense here as it makes the whole object optional.

**Review by axisiscool — DISMISSED** [review]

File: src/util/Types.ts (hunk 1 above)
Comment (PalmDevs) [reply src/util/Types.ts]: It does actually, as all properties are not required. In the options, you'll have to supply `null`s, which is annoying.

Comment (PalmDevs) [reply src/util/Types.ts]: You'll have to supply 20+ `null`s, as you can see here:
image

Comment (axisiscool) [reply src/util/Types.ts]: > You'll have to supply 20+ `null`s, as you can see here:
> 
> image

The embed typings are wrong. I'll approve with `Partial` for now but embeds need certain items like `title`.

**Review by axisiscool — APPROVED** [review]

**Comment by axisiscool** [comment]
Turns out I need your review. I'll follow this up with more type fixes later.

**Review by knokbak — APPROVED** [review]

**Comment by axisiscool** [comment]
@PalmDevs Let me know if this fixes your issue referenced in the review above.

**Review by knokbak — APPROVED** [review]

**Comment by knokbak** [comment]
i should add: remember to always add typedoc comments so the docs can be generated correctly

**Comment by PalmDevs** [comment]
@axisiscool `fields` would still be required in the options.
Embeds constructed with MessageEmbed works fine but from the raw object (which will be implemented soon) still require you to put fields as an array.
The hacky fix is just to make all properties optional for `MessageContents.embeds` but a better way is to make Typescript require one of these properties `title`, `description`, or `fields` with atleast one field element inside.

**Comment by axisiscool** [comment]
> @axisiscool `fields` would still be required in the options.
> 
> Embeds constructed with MessageEmbed works fine but from the raw object (which will be implemented soon) still require you to put fields as an array.
> 
> The hacky fix is just to make all properties optional for `MessageContents.embeds` but a better way is to make Typescript require one of these properties `title`, `description`, or `fields` with atleast one field element inside.

`fields` is defined as a blank array so you wouldn't have to define it anywhere.

**Comment by axisiscool** [comment]
yolo

**Pull request merged**